docs/explanation/remote-build.rst

```diff
@@ -84,37 +85,43 @@ Current
 ``--platform`` and ``--build-for``
 **********************************
 
-If  ``--platform`` or ``--build-for`` are provided, Snapcraft will:
-
-#. parse the project metadata for ``platforms`` or ``architectures`` keywords
-#. create a build plan
-#. filter the build plan with whichever of ``--build-for`` or ``--platform``
-   was specified
+``--platform`` or ``--build-for`` can only be provided when the ``platforms``
```

Comment:
I guess I'm missing something here, but how can the user use a "--platform" to select what to build if the project has no platforms? Is it the same semantics as "--build-for" in this case? i.e., possible values are debian architectures and the build will happen on a host of that architecture in launchpad?

Reply:
It has the same semantics as `--build-for`.  This is a step towards https://github.com/canonical/snapcraft/issues/4992.  I've made the behavior more clear in the documentation and in that issue.